### PR TITLE
Adds OmegaStation back into rotation.

### PR DESCRIPTION
**Pull request by sergeirocks100**
## About The Pull Request

This, well, readds OmegaStation back to the game. I've spent the past month updating it, most of the excluded areas and all of the missing non-head jobs have been added in, many of the issues with the map have been fixed, and i think it's more or less ready for rotation.

Here's a screenshot:[https://amel.pw/trevie/screenshots/s8.png](https://amel.pw/trevie/screenshots/s8.png)

Credits go to me, of course, Okand and whoever else contributed to the map, and the developers of Skyrat's version of Omega, for giving me ideas on where to put things as well as the Toxins Lab and Quartermaster's Office copypasted from their version of the map.

I highly recommend that this gets test merged first, as there might be bugs with the map that need to be ironed out.

## Why It's Good For The Game

Right now, i don't think /tg/ has any real lowpop maps. Pubby is the only "low-pop" map in rotation, and not only is it as large as a medium-pop map, it doesn't really serve low population counts any better than, for example, Meta does.

Omega is more tailored for low population counts than Pubby is, and i think it will do well on low population servers such as Campbell, small or private servers, or during graveyard hours on more populated servers.

## Changelog
:cl:
add: NT's decommissioned "Omega" line of stations have been put back into use.
/:cl:

### CODE REVIEW

**Comment by Ghilker** [comment]
you have to remake the piping to layer2 and layer3 for distro and waste as all our maps have now

**Comment by Rohesie** [comment]
Very nice

**Comment by wesoda25** [comment]
Will you be maintaining this map in the future?

**Comment by Fikou** [comment]
missing the anomaly compressor

File: _maps/map_files/OmegaStation/job_changes.dm [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+/datum/job/clown/New()
+	..()
+	MAP_JOB_CHECK
+	supervisors = "nobody but yourself" //Honk
+
+MAP_REMOVE_JOB(hos)
+MAP_REMOVE_JOB(chief_engineer)
+MAP_REMOVE_JOB(cmo)
+MAP_REMOVE_JOB(rd)
```
Comment (Fikou) [inline _maps/map_files/OmegaStation/job_changes.dm:143]:
```suggestion
MAP_REMOVE_JOB(rd)

```
trailing newline

File: code/modules/jobs/map_changes/map_changes.dm [hunk 2]
```diff
@@ -5,3 +5,6 @@
 
 #include "..\..\..\..\_maps\map_files\IceBoxStation\job_changes.dm"
 #undef JOB_MODIFICATION_MAP_NAME
+
+#include "..\..\..\..\_maps\map_files\OmegaStation\job_changes.dm"
+#undef JOB_MODIFICATION_MAP_NAME
```
Comment (Fikou) [inline code/modules/jobs/map_changes/map_changes.dm:10]:
```suggestion
#undef JOB_MODIFICATION_MAP_NAME

```
ditto

**Comment by kriskog** [comment]
You need to make sure it works with #46956, or look into having that fixed. You say you did the missing head roles, just make sure any other missing jobs are actually working. There used to be several issues related to omegastation and missing roles that were closed with its removal.

**Comment by Fikou** [comment]
some rooms like the diner, toxins and experimentor feel really weird and empty
xenobio's assymetry is really unepic

**Comment by ArcaneMusic** [comment]
You're merging straight from your master :eyes:

**Comment by GravitronBlue** [comment]
From the screenshot the two right side xenobiology pens appear to lack disposal bins for the disposal outlets in them.

**Comment by sergeirocks100** [comment]
Sorry for the lack of replies, i was staring at the inside of my own eyelids.

I plan on redoing the piping in the future, but it would have been a real pain in the ass to do, and the current piping does the job, so this is something i've put off for now.

I do plan on maintaining this map if it gets merged into the codebase.

I don't know what a "anomaly compressor" is, i'm not sure why i would need cytology components when, as far as i know, you can print those at the protolathe, i'll see what i can do about the lack of spawn points.

I had no problems with late joining during my testing of the map, so i think i'm fine.

I do agree that some areas look a little plain, but i don't really know how i would address that issue, i'll try to fix Xenobio's two lopsided pens as well as the disposal chute issues with them.

**Comment by Fikou** [comment]
/obj/machinery/research/explosive_compressor
place it in toxins

there should be one cytology locker somewhere in xenobio

**Comment by sergeirocks100** [comment]
done, and done

**Comment by FloranOtten** [comment]
YES! FINALLY! BLESSED MAP RETURNS

**Comment by TheVekter** [comment]
Maybe extend Xenobio a little? Needs more room, more pens.

**Comment by sergeirocks100** [comment]
I expanded it out to 7 pens, i probably won't be expanding it more after this, though.

**Comment by uomo91** [comment]
I got some runtimes on this, one related to radio stuff, one related to lattices.

**Comment by uomo91** [comment]
Also, any chance of a public mining shuttle getting added?

**Comment by sergeirocks100** [comment]
I'm not very experienced with SS13 mapping, but i'll see if i can do something about the runtimes.

And while a public mining shuttle may be added in sometime, right now, i have no clue where i would put it.

**Comment by Rohesie** [comment]
By a headcoder decree, we cannot add more maps without removing existing ones.
A lowpop map is good, and with some changes and fixes this one looks promising.
IceBox or Pubby, which one goes?

**Comment by TheVekter** [comment]
That seems odd, but y'all probably know something I don't.

Should put it to a vote I guess?

**Comment by uomo91** [comment]
psych

Would you consider redoing the seats and table in the Psych's office like this?
Only with the couch on the north wall is it possible for patients to lay on the couch, which I thought was really funny.

**Comment by sergeirocks100** [comment]
I think that we should put it to a vote as well. Although if i got to pick, i would probably pick Pubby. Ice Box is too much of a interesting idea to remove, and Pubby, as much as i like it, isn't all that great for lowpop in my opinion.

I'll go ahead and rearrange the Psychologist's Office's couches.

**Comment by uomo91** [comment]
@sergeirocks100 I've fixed all the runtimes I've found, I'll PR to your branch shortly. Merge it if you'd like.

There were some doubled up catwalks on the solar, the AI intercomms had some funky var probably left over from oldcode or something, and some APCs were misnamed and had incorrect area strings.

**Comment by sergeirocks100** [comment]
Thanks.

**Comment by uomo91** [comment]
https://github.com/sergeirocks100/tgstation/pull/1

**Comment by wesoda25** [comment]
Fact of the matter is there’s a reason pubby is in the game and omega isn’t

**Comment by LemonInTheDark** [comment]
Mapmerge the shuttle yeah?

**Comment by uomo91** [comment]
@sergeirocks100 I opened another PR, I fixed up the solar (no longer roundstart wired, has some spaces added so wiring doesn't look goofy), solar maint power stuff, and removed the rapid pipe cleaner layers from engineering since they're not useful and were taken out of other maps.

**Comment by sergeirocks100** [comment]
Again, thank you.

**Comment by uomo91** [comment]
No problem.
I'm noticing some pretty big sections of wiring that are messed up or missing, look at nanite and experimentor rooms for instance.
Port maint seems to totally lack an APC.

**Comment by sergeirocks100** [comment]
To be honest, i think i forgot to add wiring to that area, i'll do that right away.

**Comment by uomo91** [comment]
I'm also seeing some disconnected areas sharing area types, mostly maint ones.
You shouldn't do that, give each physically distinct area its own APC.

I'll try to give fixing more of this stuff a go tomorrow maybe.

**Comment by cacogen** [comment]
I hate more

**Comment by cacogen** [comment]
We don't need more maps. We need less maps that are maintained and iterated on to a higher standard.

**Comment by ShadowmanV92** [comment]
Nobody likes pubby, all my homies hate pubby. Omega is at least a breathe of fresh air and much more of a lowpop map then pubby ever was

**Comment by uomo91** [comment]
I just opened a new PR that fixed a lot of powernet things.

- Fixed AI sat SMES and wiring.
- Got rid of a duplicate APC in prison.
- Two random misplaced power cables deleted.
- There was a virologist spawn landmark in the SM, yikes. No longer.
- Properly connected the engineering SMES to the grid.
- Tried to add more redundancy to the station's wiring layout so there aren't a bunch of huge points of failure that wipe out half the station's power with one wire snip.
- Fixed some more wiring and APC placement in sci and maint.
- Properly set up the turbine and connected it to a SMES and the grid.
- Same thing with the powernet fixes, for the gravgen.
- Cleaned up a couple assorted small things I found like item placement in maint.

**Comment by uomo91** [comment]
Also eventually, probably sooner rather than later, you're going to need to turn this into TGM format.
Look into the map editor called StrongDMM, if you're not already using it.

**Comment by sergeirocks100** [comment]
Again, thank you for helping me out with my map.

I have been using StrongDMM for the map, but i don't know how to export to TGM.

**Comment by uomo91** [comment]
> Again, thank you for helping me out with my map.

Very happy to help, I love Okand's work.

> I have been using StrongDMM for the map, but i don't know how to export to TGM.

File > Preferences > Save Options > Map Save Format > then switch from provided or Byond, to TGM.

**Comment by uomo91** [comment]
I opened another PR mostly fixing things I messed up.
The wiring and SMES placement in the gravgen is better now, the SM is isolated from the main powernet now so it won't fry anyone who gets shocked, and some missing plating under the windows near the solars was added.

**Comment by uomo91** [comment]
Atmos seems to lack a mixer, and the turbine has a "incinerator chamber output inlet" instead of a scrubber inside... I don't play a lot of atmos so I'm not sure what to do to remedy the above, if anything.

**Comment by Kylerace** [comment]
toxins doesnt have enough canisters, so i made a pr to your branch to make it have 4 o2 canisters and 4 plasma by making the toxins storage room slightly bigger. i like the layout of this station but i have some concerns. i dont like how atmospherics is the same size as the rest of engineering combined. in fact the atmospherics room in this station is even bigger than metas' atmos room, despite meta being about twice the size of this map. im not complaining that atmos is huge, but it doesnt make sense for it to be this size compared to everything else. honestly the bar takes up a ton of space too. i think that itd be best if vault and the ai upload were split up so that upload could have turrets, i think you could fit one of them in this little maint space.
Screenshot_211
there should probably be another solar on the opposite end of the station although i dont think its a big problem, and for some reason there seems to be a beaker of sulfuric acid in science on the circuit fabricator.

however a possibly big flaw i see so far with the station is that its scrubber and air supply pipes arent in different layers like the rest of the maps. uh, im not 100% sure if that will actually end up being a problem later if a maintainer refuses to merge it because of it. Anyways, i wish you best of luck and hopefully you can put icebox or pubby in the fucking garbage.

**Comment by sergeirocks100** [comment]
Thanks for helping with the map.

I'm not sure what i can do about the oversized atmos and the upload, but i'll look into it.

Omega long predates the great repiping, i do plan on repiping it one day if it gets merged. But it would be a pain in the ass to replace them all right now, and the existing pipes do the job just fine, so i'm leaving it alone for now.

**Comment by 81Denton** [comment]
@tgstation/commit-access What's the consensus on this considering there was a recent vote that pruned the amount of maps we have? I don't want @sergeirocks100 to waste his time.

**Comment by Rohesie** [comment]
This would have to be a replacement for an existent map ([source](https://sb.atlantaned.space/info/mapularity)).
Currently Pubby is the least played map, and I'd be up for making the switch to Omega, for variety if nothing else.

**Comment by Jared-Fogle** [comment]
Agreed with @Rohesie--if Pubby is meant to be the low pop map, then Omega should replace it.

**Comment by 81Denton** [comment]
I'm more partial towards Pubby because it doesn't suffer from the explosion issue that Omega suffers from (maxcaps can make entire areas uninhabitable due to its small size). I think that adding Omega for "extreme" lowpop with a very low pop limit on top of the existing maps would work well too.

**Comment by Jared-Fogle** [comment]
I'm personally opposed to adding a new map without replacing another one just for the sake of maintainability.

**Comment by Rohesie** [comment]
I'd rather we had as many maps as decent and maintained versions exist, and this would imply keeping the existing 4 maps and adding OmegaStation and ShipStation on top of them. I don't feel like 6 maps is unreasonable, and each offers something different.

That said, oranges wants us to move towards less maps, ideally two, so we have to make choices.
I'd rather we saw variety and rotation and were not forever stuck on the same maps, myself. I like Pubby and feel it can return in t he future, but I'd favor change (when the incoming map is good) over just staying as we are.

**Comment by optimumtact** [comment]
The consensus is that my decree is the same as it has been for some time.

If you want new maps in the codebase you need to replace an existing one.

**Comment by EOBGames** [comment]
Hello! I've run your map over with the debug tools to check for any issues. Here's what I've found:
Camera Debugger:
cameradebugoutput You have a few matching c-tags, this will cause issues for viewing the cameras via a console. Resolve these by editing the c-tag variable on the offending cameras to something unique.

Atmos Debugger:
pipenetdebugouput
There are quite a few disconnected atmospherics pipes, mostly in the brig near the entrance to permabrig. This is an easy fix, just connect up the missing pipes here:
image
and here:
image
Additionally, an air vent is disconnected here, in toxins mixing:
image
On a personal note for atmos, I believe the map could benefit from a 2-layered atmospherics system, as is now more standard on our current maps. This would reduce the number of pipes in walls (which causes issues for repairs) and generally simplify the layout of the pipenet.

Powernet debugger:
powernetdebugoutput
Not much here! This cable in the brig is doubled, please remove one of the cables:
image
The AI satellite being unwired is, I believe, intentional.

There are several doubled APCs: the fore primary hallway, aft primary hallway, and starboard primary hallway all have two APCs each. The port maint also has two, one below virology, and one to the left of the library. Remove one or the other.

The camera coverage in the far east of science is rather spotty, particularly in the hallway between genetics and the servers. I would strongly recommend adding a camera somewhere around the door to genetics.

Hydroponics has no APC. Unless you want sad botanists, I'd add one.

Permabrig also has no APC. Unless you want a riot on your hands, I'd add one.

Toxins Storage has no APC, again I'd add one- generally all station areas should have an APC. This area is also lacking an air alarm, which should also probably be added.

Tech storage has no air alarm.

Several rooms to the right side of science have no intercoms, particularly genetics.

The Medbay APC is in an area that is inaccessible to anyone without medbay access- this isn't ideal for engineers fixing it. Perhaps shift it elsewhere in the room, such as next to the door to virology?

Personal observation: I notice that the map has no disposals system. Adding this in could be a good goal to improve the map and help further advance it from the original version, if you're feeling up to it.

For future reference, the debug tools can be found via activating debug verbs in the debug tab, then within the mapping tab that appears. These really help for finding hard to spot problems with your map, such as the ones listed above.

**Comment by sergeirocks100** [comment]
Thanks for informing me of these issues, i'll try to fix them right away.

**Comment by Cyberboss** [comment]
> The more advanced version of this is a seperate repo, which allows map creators to completely skip the the review cycle and gives them more freedom, but reduces the chances they will get normal contributor help updating their map.
> 
> Until then the process is unlikely to change unless there are new headcoders.

I'm fully in agreement with this. Which is why it's been a longstanding goal of mine to push out support for maps that can be remotely downloaded with a versioning system, pushing that tax onto people that want their map to be played.

Unfortunately, life has been getting in the way of me setting it up.

**Comment by Cyberboss** [comment]
I have the overall steps outlined for that. If anyone wants to help, ping me on discord.

**Comment by SpaceManiac** [comment]
> I have the overall steps outlined for that. If anyone wants to help, ping me on discord.

Is writing them down that hard

**Comment by Cyberboss** [comment]
Started some work in #53663 basically the last thing I need is a way to download and unzip .zips from GitHub via rust-g. I asked @spookydonut to potentially help with that

**Comment by Jared-Fogle** [comment]
White ships are randomized now--please don't add another extremely small white ship since they are no longer map specific.

**Comment by sergeirocks100** [comment]
I realized that too, which is why i will probably be removing Omega's whiteship dock for now, as there isn't a single white ship in the game right now besides Omega's own that can fit in the tiny space.

File: html/changelog.html [hunk 3]
```diff
@@ -51,6 +51,25 @@
...
+				<li class="tweak">As a ghost, you will not see emotes from mobs that dont have a client and are out of your range, while using ghost sight</li>
+				<li class="tweak">Messages based on sight aren't obscured by darkness if you're right next to the source</li>
+				<li class="rscadd">Added a rest keybind! The default hotkey is U</li>
+			</ul>
+			<h3 class="author">Timberpoes updated:</h3>
+			<ul class="changes bgimages16">
+				<li class="bugfix">Apple Cakes no longer slice into Slime Cake Slices</li>
+				<li class="bugfix">Minor runtime with no gameplay impact related to printing implant cases.</li>
+				<li class="bugfix">Non-gameplay impacting runtime on holopads when the user spoke after placing a call but before it has been connected.</li>
+				<li class="bugfix">EMP-shielded cameras will no longer runtime when destroyed.</li>
```
Comment (optimumtact) [inline html/changelog.html:70]: makes you cringe

**Comment by SpaceManiac** [comment]
I'm closing this for now in honor of our standing policy against new maps in the codebase.

I recommend keeping track of or getting involved in the project to support out-of-codebase maps.

You should avoid making PRs from your fork's "master" branch, as our changelog cronjobs will run there and cause conflicts.

**Comment by TheVekter** [comment]
I... okay? What process needs to be followed to replace one of the maps currently implemented in the codebase?

**Comment by sergeirocks100** [comment]
As saddened as i may be at this PR getting closed, i understand your point.

I don't like to make a new branch for every PR, as i have to download over 1.5 gigabytes worth of stuff every time i make a fork, and that's pretty unbearable on my slow DSL connection.

I had no idea that there was a policy against adding in new maps period, and if i had known this from the start, i probably wouldn't have bothered with this at all.

I will still be keeping this map up to date, however, and maybe a few months from now, i'll PR it again.

**Comment by SpaceManiac** [comment]
I like OmegaStation as a map, but our team's circumstances are such that we don't want to maintain more than the 4 maps we already are. Maybe this will change in the future.

> I don't like to make a new branch for every PR, as i have to download over 1.5 gigabytes worth of stuff every time i make a fork, and that's pretty unbearable on my slow DSL connection.

Redownloading everything is absolutely not necessary to use a different branch name than `master` for your PRs. Our [Setting up git](https://tgstation13.org/wiki/Setting_up_git) guide has some details, and you can read the [Git manual](https://git-scm.com/doc) for more.

In particular the section describing `git checkout -b <BRANCHNAME>` is relevant.

**Comment by sergeirocks100** [comment]
You do know that Omega was going to replace another map, right?

**Comment by SpaceManiac** [comment]
"going to replace another map" is not a path to mergeability because it does not concretely specify _which_ other map would be replaced.

After discussion with @Timberpoes, you are welcome to open a new PR if you both:

1. Title the PR something like "Replace PubbyStation with OmegaStation", since that's what would have to happen for the PR to be merged.
    - It's up to you whether you want to remove PubbyStation in the PR itself or leave that step to us (maintainers) to handle.
2. Use a branch name other than `master`. This is important for technical reasons and, as I said before, *does not* require multiple redownloads of the repository.

If you're having trouble managing the branches, ask for help on coderbus.

**Comment by sergeirocks100** [comment]
Thank you, i'm going to go ahead and do just that after i manage to fix up my repository.

**Pull request closed without merging**